src/Microsoft.DotNet.Tools.Test/TestRunners/AssemblyTestRunnerResolver.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Tools.Test
+{
+    public class AssemblyTestRunnerResolver : ITestRunnerResolver
+    {
+        private readonly string _directoryOfAssemblyUnderTest;
+
+        private readonly IDirectory _directory;
+
+        public AssemblyTestRunnerResolver(string assemblyUnderTest) :
+            this(assemblyUnderTest, FileSystemWrapper.Default.Directory)
+        {
+        }
+
+        internal AssemblyTestRunnerResolver(string assemblyUnderTest, IDirectory directory)
+        {
+            _directoryOfAssemblyUnderTest = directory.GetDirectoryFullName(assemblyUnderTest);
+            _directory = directory;
+        }
+
+        public string ResolveTestRunner()
+        {
+            var testRunnerPath = _directory.GetFiles(_directoryOfAssemblyUnderTest, "dotnet-test-*").FirstOrDefault();
```

Comment:
If this returns default, you may get a null ref on the next call to Path.GetFileNameWithoutExtension